### PR TITLE
Message new send composition

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [19853](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/19853)
- [19849](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/19849)

### Description

Add node-tap to sdk package.
Global .tap in .gitignore
Replaced old POST /messages with new as per design
Commented out the "by template" endpoint
Moved

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

**Review by Thurfjell — COMMENTED** [review]
Add OK Created response to POST /mesages - done

File: apps/messaging-api/types/schemaDefinitions.ts [hunk 1]
```diff
@@ -154,3 +154,28 @@ export const MessageEvent = Type.Array(
     createdAt: Type.String({ format: "date-time" }),
   }),
 );
+
+export const MessageCreate = Type.Object({
+  preferredTransports: Type.Array(
+    Type.Union([
+      Type.Literal("email"),
+      Type.Literal("sms"),
+      Type.Literal("lifeEvent"),
```
Comment (Thurfjell) [inline apps/messaging-api/types/schemaDefinitions.ts:163]: @relvao Do we include life event as a valid transport at this stage?

File: apps/messaging-api/types/schemaDefinitions.ts [hunk 2]
```diff
@@ -154,3 +154,28 @@ export const MessageEvent = Type.Array(
...
+    ]),
+  ),
+  userId: Type.String(),
+  security: Type.String(), // what do we wana roll.. string?
```
Comment (Thurfjell) [inline apps/messaging-api/types/schemaDefinitions.ts:167]:
```suggestion
  security: Type.String(), 
```

File: apps/messaging-api/types/usersSchemaDefinitions.ts [hunk 3]
```diff
@@ -212,6 +212,14 @@ export const RecipientSchema = Type.Object({
   phoneNumber: NullableStringType,
   emailAddress: NullableStringType,
   birthDate: NullableStringType,
-  preferredTransports: Type.Array(Type.String()),
+  preferredTransports: Type.Array(
+    Type.Union([
+      Type.Literal("email"),
+      Type.Literal("sms"),
+      Type.Literal("lifeEvent"),
+    ]),
```
Comment (Thurfjell) [inline apps/messaging-api/types/usersSchemaDefinitions.ts:220]: Let's create a type for this and re-use

File: apps/messaging/app/[locale]/admin/send-a-message/[action]/ScheduleForm.tsx [hunk 4]
```diff
@@ -24,89 +24,125 @@ export default async (props: MessageCreateProps) => {
...
+            ppsn: user.ppsn || "",
+          },
         );
+      } catch (err) {
+        // Needs redundancy strategy here
+        continue;
       }
+
+      try {
+        const preferredTransports: ("sms" | "email" | "lifeEvent")[] = [];
```
Comment (Thurfjell) [inline apps/messaging/app/[locale]/admin/send-a-message/[action]/ScheduleForm.tsx:101]: Check if type is warranted

File: apps/messaging/messages/en.json [hunk 5]
```diff
@@ -163,7 +163,8 @@
     "SuccessForm": {
       "title": "Your message has been sent",
       "body": "A copy of this message will appear in your sent messages.",
-      "submitText": "Send another message"
+      "submitText": "Send another message",
+      "partialSuccessError": "Sorry, we were unable to send your message to a number of recipients"
```
Comment (Thurfjell) [inline apps/messaging/messages/en.json:167]: Check with Rich for copy.

File: apps/profile-api/routes/user/index.ts [hunk 6]
```diff
@@ -285,35 +286,36 @@ export default async function user(app: FastifyInstance) {
...
           }),
-          404: Type.Null(),
+          "4xx": HttpError,
+          "5xx": HttpError,
         },
       },
     },
     async function handler(request) {
+      const ids = request.body.ids;
+      console.log("asking for ", ids);
```
Comment (Thurfjell) [inline apps/profile-api/routes/user/index.ts:301]:
```suggestion
```

File: apps/profile-api/routes/user/index.ts [hunk 7]
```diff
@@ -322,16 +324,20 @@ export default async function user(app: FastifyInstance) {
...
-
-        if (!users.length) {
-          throw new NotFoundError(ERROR_PROCESS, "User not found");
-        }
-
-        return { data: users };
+        users.push(...usersQueryResult.rows);
       } catch (err) {
-        return { error: err };
+        console.log("Krashar vi direkt haer eller? ", err);
```
Comment (Thurfjell) [inline apps/profile-api/routes/user/index.ts:329]:
```suggestion
```

**Review by edge33 — APPROVED** [review]
LGTM

**Pull request merged**